File: relmgr_java/RelationshipManager-Java/src/relationshipmanager/turbo/IRelationshipManager.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package relationshipmanager.turbo;

import java.util.List;

/**
 *
 * @author Tarik
 */
public interface IRelationshipManager {
    public void AddRelationship(Object fromObj, Object toObj, String relId);
    public Object FindObjectPointedToByMe(Object fromObj, String relId);
    public List FindObjectsPointedToByMe(Object fromObj, String relId);
    public void RemoveRelationship(Object fromObj, Object toObj, String relId);
    public void RemoveAllRelationshipsInvolving(Object obj, String relId);
    public Object FindObjectPointingToMe(Object toObj, String relId);
    public List FindObjectsPointingToMe(Object toObj, String relId);
    public int Count();
    public int CountRelationships(String relId);
    public void Clear();
    public void EnforceRelationship(String relId, Cardinality cardinality);
    public boolean DoesRelIdExistBetween(Object fromObj, Object toObj, String relId);
    public List FindRelIdsBetween(Object fromObj, Object toObj);
    public String Dump();
}

File: relmgr_dotnet/csharp/Relationship Manager Interfaces/IRelationshipManager.cs
using System;
using System.Collections.Generic;
using System.Collections;
using System.Text;

namespace RelationshipManager.Interfaces
{
    public enum Cardinality
    {
        OneToOne,
        OneToMany,
        ManyToOne,
        ManyToMany
    }

    public enum Directionality
    {
        UniDirectional,
        DirectionalWithBackPointer,
        DoubleDirectional
    }

    public interface IRelationshipManager
    {
        void AddRelationship(object fromObj, object toObj, string relId);
        void AddRelationship(object fromObj, object toObj);
        //void EnforceRelationship(string relId, string cardinality);
		//void EnforceRelationship(string  relId, string cardinality, string directionality);
        void EnforceRelationship(string relId, Cardinality cardinality);
        void EnforceRelationship(string relId, Cardinality cardinality, Directionality directionality);
        IList FindObjectsPointedToByMe(object fromObj, string relId);
		object FindObjectPointedToByMe(object fromObj, string relId);
	 	IList FindObjectsPointingToMe(object toObj, string relId);
		object FindObjectPointingToMe(object toObj, string relId);
		void RemoveRelationship(object fromObj, object toObj, string relId);
		void RemoveAllRelationshipsInvolving(object obj, string relId);
        int Count();
        int CountRelationships(string relId);
        void Clear();
        bool DoesRelIdExistBetween(object fromObj, object toObj, string relId);
        IList FindRelIdsBetween(object fromObj, object toObj);
    }

    public interface IRM
    {
        object B(object toObj, string relId);
        System.Collections.IList BS(object toObj, string relId);
        void ER(string relId, Cardinality cardinality);
        void ER(string relId, Cardinality cardinality, Directionality directionality);
        void NR(object fromObj, object toObj, string relId);
        void NRS(object obj, string relId);
        object P(object fromObj, string relId);
        System.Collections.IList PS(object fromObj, string relId);
        void R(object fromObj, object toObj, string relId);
        bool QR(object fromObj, object toObj, string relId);
        int Count();
        int CountRelationships(string relId);
        void Clear();
    }
}
